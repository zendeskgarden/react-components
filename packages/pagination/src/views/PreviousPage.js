import PropTypes from 'prop-types';
import styled from 'styled-components';
import PaginationStyles from '@zendesk/garden-css-pagination';
import { retrieveTheme } from '@zendesk/garden-react-theming';

import { version } from '../../package.json';
import Page from './Page';
const COMPONENT_ID = 'pagination.previous_page';

/**
 * Accepts all `<li>` props
 */
const PreviousPage = styled(Page).attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': version,
  className: PaginationStyles['c-pagination__page--previous']
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

PreviousPage.propTypes = {
  current: PropTypes.bool,
  focused: PropTypes.bool,
  hovered: PropTypes.bool,
  hidden: PropTypes.bool
};

/** @component */
export default PreviousPage;
