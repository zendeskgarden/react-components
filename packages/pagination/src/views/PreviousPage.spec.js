import React from 'react';
import { shallow } from 'enzyme';

import PreviousPage from './PreviousPage';

describe('PreviousPage', () => {
  it('renders default styling', () => {
    const wrapper = shallow(<PreviousPage />);

    expect(wrapper).toMatchSnapshot();
  });
});
