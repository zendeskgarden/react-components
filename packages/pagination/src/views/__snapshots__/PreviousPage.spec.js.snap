// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviousPage renders default styling 1`] = `
<Page
  className="c-pagination__page--previous "
  data-garden-id="pagination.previous_page"
  data-garden-version="version"
/>
`;
